### ISSUE TITLE
"ERROR: argument of AND must be type boolean, not type name" when attempting to expand "Data types" in DBeaver 21.0.3

### ISSUE THREAD

**Issue by lorenzo-ghub** [dbeaver/dbeaver#12186]
#### System information: 
- Operating system (distribution) and version: Windows 10 Pro, 20H2, OS build 19042.870
- DBeaver version: DBeaver 21.0.3
- Additional extensions: none

#### Connection specification:
- Database name and version: PostgreSQL 12.6
- Driver name: org.postgresql.Driver
- Do you use tunnels or proxies (SSH, SOCKS, etc)? no

#### Describe the problem you're observing:
Whenever attempting to expand the "Data types" folder in DBeaver, I receive the following error:
```
SQL Error [42804]: ERROR: argument of AND must be type boolean, not type name
  Position: 501
  ERROR: argument of AND must be type boolean, not type name
  Position: 501
```
![DBeaver-DataTypes-Error](https://user-images.githubusercontent.com/9766712/115453379-91c72280-a1ed-11eb-9515-7cd06c68ad93.png)

#### Steps to reproduce, if exist:
1. Open any database
2. Navigate to:
    - Schemas
    - public
3. Attempt to navigate into "Data types"
4. Error is thrown as noted above.

#### Include any warning/errors/backtraces from the logs
[dbeaver-debug.log](https://github.com/dbeaver/dbeaver/files/6346224/dbeaver-debug.log)
[debeaver.log](https://github.com/dbeaver/dbeaver/files/6346227/debeaver.log)

**Comment by LonwoLonwo** [comment]
Thanks for the bug report!

**Comment by LonwoLonwo** [comment]
Reproducible for those schemas that do not have data types.

**Comment by LonwoLonwo** [comment]
Fixed in DBeaver 21.0.4

**Comment by uslss** [comment]
verified
